gobblin-metastore/src/main/java/org/apache/gobblin/metastore/MysqlStateStore.java

```diff
@@ -73,7 +74,7 @@
  *
  * @param <T> state object type
  **/
-public class MysqlStateStore<T extends State> implements StateStore<T> {
+public class MysqlStateStore<T extends State> implements StateStore<T>, DatasetStateStore<T> {
```

Comment:
This is looking strange. The child class MySqlDatasetStateStore extends MySqlStateStore and implements DatasetStateStore. And the parent class is implementing the interface. See my comment earlier about considering the possibility of MysqlJobStatusStore extending MySqlDatasetStateStore.

Reply:
Oh, that is not done intentionally, after making this class implement DatasetStateStore, I forgot to to make its child class to not implement DatasetStateStore